pyteal/compiler/optimizer/optimizer.py

```diff
@@ -0,0 +1,72 @@
+from dataclasses import dataclass
+from typing import List, cast
+
+from pyteal.ir.tealcomponent import TealComponent
+from pyteal.ir.tealop import TealOp
+from pyteal.ir.ops import Op
+
+
+@dataclass
+class OptimizeOptions:
+    """An object which specifies the optimizations to be performed.
+    Args:
+
+        use_slot_to_stack (optional): cancel contiguous store/load operations
+            that have no load dependencies elsewhere.
+        iterate_optimizations (optional): repeat the optimizations until the
+            code is no longer changed."""
+
+    use_slot_to_stack: bool = False
```

Comment:
I have two suggestions:
* Can you give `use_slot_to_stack` a more general name? In the future it might not only move variables from scratch slots to the stack, it could also reduce the number of loads and stores or do something more advanced. Maybe just `scratch_slots`? How optimizations will evolve is a bit of an open question, but right now I think `OptimizeOptions` should only have fields about general categories of optimizations, and in the future when we have more specific options for each category, we might want to change the value from a boolean to _another_ options objects just for that category of optimizations.

* Why should `iterate_optimizations` be exposed? Is there a scenario where you would want to disable this?

Reply:
I agree with the first point. In regards to the second, this was mainly a conservative decision to allow for fine grained control in potential cases where iteration would be undesirable, although I don't know of any at the moment. I realize that trying to plan for all contingencies isn't the best argument so I don't mind removing the flag.